client/engine/engine.go

```diff
@@ -264,6 +265,18 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Object
 	// Probably should have a better check that only adds it to CompletedObjectives if it was completed in this crank
 	if waitingFor == "WaitingForNothing" {
 		outgoing.CompletedObjectives = append(outgoing.CompletedObjectives, crankedObjective)
+
+		// Whenever a direct funding objective completes we want to create a consensus_channel
+		if dfo, isDfo := crankedObjective.(*directfund.Objective); isDfo {
+			c, err := consensus_channel.CreateFromDirectFundingObjective(*dfo)
+			if err != nil {
+				return ObjectiveChangeEvent{}, fmt.Errorf("could not create consensus channel for objective %s: %w", objective.Id(), err)
+			}
+			err = e.store.SetConsensusChannel(c)
+			if err != nil {
+				return ObjectiveChangeEvent{}, fmt.Errorf("could not store consensus channel for objective %s: %w", objective.Id(), err)
+			}
+		}
```

Comment:
Could this be factored into a function?

Reply:
And should we do this _before_ we do `e.executeSideEffects(sideEffects)`? I think this would help to reduce the chance of a race between an incoming message  from a counterparty which is triggered by `e.executeSideEffects`, and the insertion of the `ConsensusChannel` into the store.